### PR TITLE
Add 'Debug' option for API requests in Apple News to output debugging info

### LINKED ISSUE #1110: Add Debugging Mechanism for Apple News API Requests
### Description

Implement a debugging mechanism in the Apple News plugin that allows developers to easily preview the full API request data for a given post. The debugging feature should be easily accessible, but restricted to admin users only for security.

1. Add a new `debug_api()` method to the `Apple_Push` class that outputs the API request data when a specific URL parameter (`apple_news_debug`) is present and the current user is an admin.
2. The `debug_api()` method should output the post ID for context, followed by the JSON article data, asset bundles, and post metadata. The output should be clearly separated and formatted for readability.
3. After outputting the debug data, the method should exit to prevent further execution and avoid actually sending the API request.
4. Add a new "Debugging" section to the plugin that explains how to use the `apple_news_debug` parameter to preview API request data.

To test this feature:
- Verify that the debug output is only shown when the URL param is present and the user is an admin
- Confirm the output contains all expected API request data and is formatted properly
- Ensure normal API requests are not affected when the debug parameter is not set

This debugging feature will be a valuable tool for plugin developers, making it much easier to diagnose API issues and validate the data being sent to Apple News.

### Use Case

As a developer working on the Apple News plugin, I often need to see the exact data being sent to the Apple News API for a given post. This is necessary for validating the data and diagnosing any issues with the API requests. However, currently there's no easy way to inspect the full request payload, especially for data that's not included in the typical debug output, such as metadata added by custom rules.

### PR DESCRIPTION
**Pull request by attackant**
A 'Debug' link has been added to the API request in the admin section of the Apple News plugin. This link, which opens in a new tab, can be used for debugging purposes. It is particularly useful for showing the API request.

Also syncs WP scripts node dependencies with WordPress 6.5

Addresses #1110

### CODE REVIEW

File: admin/apple-actions/index/class-push.php [hunk 1]
```diff
@@ -540,16 +527,83 @@ private function process_errors( $errors ) {
...
+		if ( ! empty( $this->sections ) ) {
+			sort( $this->sections );
+			$meta['data']['links'] = [ 'sections' => $this->sections ];
+		}
+
+		// Get the various settings.
+		$is_paid         = (bool) get_post_meta( $post_id, 'apple_news_is_paid', true );
+		$is_preview      = (bool) get_post_meta( $post_id, 'apple_news_is_preview', true );
+		$is_hidden       = (bool) get_post_meta( $post_id, 'apple_news_is_hidden', true );
+		$is_sponsored    = (bool) get_post_meta( $post_id, 'apple_news_is_sponsored', true );
```
Comment (mogmarsh) [inline admin/apple-actions/index/class-push.php]: in 2.5.0, these actually can be set by the meta or inherit the channel default, so they won't always be `bool` any more. See https://github.com/alleyinteractive/apple-news/pull/1090/files

File: admin/class-admin-apple-index-page.php [hunk 2]
```diff
@@ -405,6 +407,25 @@ private function push_action( $id ) {
...
+	 *
+	 * @access private
+	 */
+	private function debug_action( ?int $id ): void {
+		$post = get_post( $id );
+		if ( ! $post ) {
+			return;
+		}
+
+		$action = new Apple_Actions\Index\Push( $this->settings, $id );
```
Comment (mogmarsh) [inline admin/class-admin-apple-index-page.php:388]: does this actually push it? Is that what we want to happen?

**Review by mogmarsh — CHANGES_REQUESTED** [review]
This looks good, but there is a change to the `is_` meta in 2.5.0 that you'll need to incorporate. Also, I had a question about if the debug actually pushes it.

Hit me up for another review. I don't want this to hold you up.

File: admin/class-admin-apple-index-page.php (hunk 2 above)
Comment (attackant) [reply admin/class-admin-apple-index-page.php:388]: No, just calls the constructor and sends in the settings, we only call the `debug` method, not the `push` method, then debug info is printed to the screen and then exits. I'll add a bit of documentation to the README.

File: admin/apple-actions/index/class-push.php (hunk 1 above)
Comment (attackant) [reply admin/apple-actions/index/class-push.php]: Great, will update.

File: admin/class-admin-apple-index-page.php (hunk 2 above)
Comment (attackant) [reply admin/class-admin-apple-index-page.php:388]: Added some docs here: https://github.com/alleyinteractive/apple-news/wiki/Usage#debug-api-requests

File: admin/apple-actions/index/class-push.php (hunk 1 above)
Comment (attackant) [reply admin/apple-actions/index/class-push.php]: Give this a look, thanks Greg.

**Review by mogmarsh — APPROVED** [review]
👍 looks good. thanks!

**Pull request merged**